samples/Jzon.cpp

```diff
@@ -25,6 +25,10 @@ THE SOFTWARE.
 #define JzonAPI __declspec(dllexport)
 #endif
 
+#ifdef WIN32
```

Comment:
This should be _MSC_VER or something. Not relevant for mingw.

Reply:
Speaking of this it would probably be a good idea to set up an msys CI

Reply:
The CI does build MinGW/msys2.  We don't have that in the nightly GHA build.  Would you like to change/fix that?

Reply:
We actually have that MSYS2 build which is triggered on PRs:
https://github.com/Exiv2/exiv2/actions/runs/1670765792

I think it is enough to have it there and not in the nightly builds. 

Now that I think about it, on Windows + MSYS2, it is ending up using MinGW (with gcc-11). The problem with the `Jzon` class is only reproducible with VisualStudio and very recent versions of clang.

Reply:
I just realized. C++17 supports CTAD. You can safely ommit the template parameters.